p2p/host/basic/basic_host.go

```diff
@@ -143,10 +150,25 @@ func NewHost(net inet.Network, opts *HostOpts) *BasicHost {
 		h.cmgr = opts.ConnManager
 	}
 
+	var relayCtx context.Context
+	var relayCancel func()
+	if opts.Relay {
+		relayCtx, relayCancel = context.WithCancel(context.Background())
+		err := circuit.AddRelayTransport(relayCtx, h, opts.RelayOpts...)
+		if err != nil {
+			// perhaps inappropriate, but otherwise we have to change the interface
+			// to return an error, which will nost likely lead to a fatality anyway
+			panic(err)
```

Comment:
The inappropriate panic was squashed in https://github.com/libp2p/go-libp2p/pull/210/commits/7f1ffcbd4e69394b9babb5dead5b940c1aaca547